app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt

```diff
@@ -206,35 +214,25 @@ class EditProfileActivity : BaseActivity(), Injectable {
         }
 
         val onBackCallback = object : OnBackPressedCallback(enabled = true) {
-            override fun handleOnBackPressed() = checkForPotentialUnsavedChanges()
+            override fun handleOnBackPressed() {
+                if (viewModel.hasUnsavedChanges(currentProfileData)) {
```

Comment:
Referencing the viewModel here is gonna trigger Lint because viewModel is a private field and this is an anonymous class so a synthetic getter is needed and Lint is programmed to hate that. ;) (that's why I created the other intermediate method).

Reply:
(any private thing from the anonymous inner class will do that)

Reply:
Ok, re-introduced that check method.

(so it can be merged)

But something is off here, at least architecturally.
(finish() in two different locations; some hassle with lifecycles there)

Reply:
Agree. Perhaps this can be improved in the future. I'll take another look now that you've 'done the bulk' of the work. Approving.

Reply:
Will continue with comments on the original PR, once this is merged. :-)

Reply:
You. The destination is your branch. :-)

Reply:
It doesn't let me merge because it says 

` /home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt:231:1 Needless blank line(s)
/home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt:45:1 Unused import
/home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt:126:108 Unnecessary long whitespace
/home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt:131:108 Unnecessary long whitespace
/home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt:147:71 Unnecessary trailing comma before ")"
/home/runner/work/Tusky/Tusky/app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt:229:27 Unnecessary trailing comma before ")"`

Are these issues in "my branch" or your changes? because the dest branch is "green".